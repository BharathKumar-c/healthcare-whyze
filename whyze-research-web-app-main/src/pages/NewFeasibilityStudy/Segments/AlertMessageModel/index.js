import React from 'react';
import { Modal } from '../../../../components/hoc';
import PropTypes from 'prop-types';
import {
  setActiveSegment,
  setSegments,
} from '../../../../redux/reducer/feasibilityStudyReducer';
import { useSelector, useDispatch } from 'react-redux';

const AlertMessageModel = (props) => {
  const { segments } = useSelector((state) => state.feasibilityStudyReducer);
  const dispatch = useDispatch();

  const handleSelectOption = () => {
    if (segments.length > 1) {
      const caseStudy = [...segments];
      caseStudy.splice(props?.activeSegment.index, 1);
      dispatch(setSegments(caseStudy));
      dispatch(
        setActiveSegment({
          name:
            props?.activeSegment.index !== 0
              ? caseStudy[props?.activeSegment.index - 1]?.name
              : caseStudy[props?.activeSegment.index].name,
          index:
            props?.activeSegment.index !== 0
              ? props?.activeSegment.index - 1
              : 0,
        }),
      );
      props.onClose(false);
    } else {
      props.onClose(false);
    }
  };

  const handleClose = () => {
    props.onClose(false);
  };
  return (
    <Modal
      title={`${props?.activeSegment.name}`}
      okButtonLabel={'Ok'}
      cancelButtonLabel={'Cancel'}
      isOpen={props.isOpen}
      onClose={handleClose}
      handleActionClick={handleSelectOption}
    >
      {segments.length > 1 ? (
        <h4>Are you sure you want to delete this segment?</h4>
      ) : (
        <h4>Unable to delete this segment. At least one segment is needed</h4>
      )}
    </Modal>
  );
};

export default AlertMessageModel;

AlertMessageModel.propTypes = {
  activeSegment: PropTypes.object,
  isOpen: PropTypes.bool,
  onClose: PropTypes.func,
};

AlertMessageModel.defaultProps = {
  activeSegment: null,
  isOpen: false,
  onClose: () => {},
};
