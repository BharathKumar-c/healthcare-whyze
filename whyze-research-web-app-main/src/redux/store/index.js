import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
