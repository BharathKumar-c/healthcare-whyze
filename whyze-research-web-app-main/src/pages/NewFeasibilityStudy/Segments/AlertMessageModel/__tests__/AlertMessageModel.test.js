import { fireEvent, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../../../redux/store';
import AlertMessageModel from '..';
import React from 'react';
function renderWithContext(element) {
  const { container, getByRole, getAllByText, getByText } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByRole, getAllByText, getByText };
}
const activeSegment = {
  name: 'demoActiveSegment',
  index: 0,
};
const activeSegmenttwo = {
  name: 'demoActiveSegment',
  index: 1,
};

describe('write testcases for Alertmessagemodal', () => {
  test('create snapshots for segments', () => {
    const { container } = renderWithContext(
      <AlertMessageModel activeSegment={activeSegment} />,
    );
    expect(container).toMatchSnapshot();
  });
  test('should  Alert modal open ', () => {
    const { container } = renderWithContext(
      <AlertMessageModel activeSegment={activeSegmenttwo} isOpen={true} />,
    );
    expect(container).toBeInTheDocument();
  });
  test('should Alert modal render content  ', () => {
    const { getByText } = renderWithContext(
      <AlertMessageModel activeSegment={activeSegmenttwo} isOpen={true} />,
    );
    const messageinModal = getByText(/Do you Want to delete these Segment/i);
    expect(messageinModal).toBeInTheDocument();
  });
  test('should Alert modal save ', () => {
    const { container, getByRole } = renderWithContext(
      <AlertMessageModel activeSegment={activeSegmenttwo} isOpen={true} />,
    );
    const submitButton = getByRole('button', { name: 'Confirm' });
    fireEvent.click(submitButton);

    expect(container).toBeInTheDocument();
  });
  test('should Alert modal close ', () => {
    const { container, getByRole } = renderWithContext(
      <AlertMessageModel activeSegment={activeSegmenttwo} isOpen={true} />,
    );
    const submitButton = getByRole('button', { name: 'Cancel' });
    fireEvent.click(submitButton);

    expect(container).toBeInTheDocument();
  });
  test('should Alert modal save with 0th index itshows message  ', () => {
    const { getByRole, getByText } = renderWithContext(
      <AlertMessageModel activeSegment={activeSegment} isOpen={true} />,
    );
    const submitButton = getByRole('button', { name: 'Ok' });
    fireEvent.click(submitButton);
    const messageinModal = getByText(/You Cannot delete these Segment/i);
    expect(messageinModal.textContent).toBe(
      'You Cannot delete these Segment . need atleast one segment',
    );
  });
});
