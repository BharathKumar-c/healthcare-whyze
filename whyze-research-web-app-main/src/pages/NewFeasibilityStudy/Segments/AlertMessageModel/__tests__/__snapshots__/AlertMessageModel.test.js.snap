// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`write testcases for Alertmessagemodal create snapshots for segments 1`] = `<div />`;
